remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala

```diff
@@ -462,23 +462,15 @@ class NettyTransport(val settings: NettyTransportSettings, val system: ExtendedA
 
   override def isResponsibleFor(address: Address): Boolean = true // TODO: Add configurable subnet filtering
 
-  // TODO: This should be factored out to an async (or thread-isolated) name lookup service #2960
-  def addressToSocketAddress(addr: Address): Future[InetSocketAddress] = addr match {
-    case Address(_, _, Some(host), Some(port)) =>
-      Future { blocking { new InetSocketAddress(InetAddress.getByName(host), port) } }
-    case _ => Future.failed(new IllegalArgumentException(s"Address [$addr] does not contain host or port information."))
-  }
-
   override def listen: Future[(Address, Promise[AssociationEventListener])] = {
     @nowarn("msg=deprecated")
     val bindPort = settings.BindPortSelector
 
     for {
-      address <- addressToSocketAddress(Address("", "", settings.BindHostname, bindPort))
+      _ <- Future.unit
```

Comment:
It no longer needs `for`, you don't even need a unit.

Reply:
That will invoke a larger range of rewrite.

Reply:
As you see, the diff is larger.